### PR TITLE
R doc 1640 patching methods

### PR DESCRIPTION
**Pull request by morhilai**

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]
LGTM, but does not parse, please fix it.

File: Documentation/4.1/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 1]
```diff
@@ -129,6 +132,25 @@ An operations interface that exposes the full functionality and allows performin
 
 {PANEL/}
 
+{PANEL: List of Script Methods}
+
```
Comment (garayx) [inline Documentation/4.1/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:136]: add link to /csharp/server/kb/javascript-engine#predefined-javascript-functions

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.java.markdown [hunk 2]
```diff
@@ -123,6 +126,23 @@ An operations interface that exposes the full functionality and allows performin
 
 {PANEL/}
 
+{PANEL: List of Script Methods}
+
+This is a list of a few of the javascript methods that can be used in patch scripts.  
+
```
Comment (garayx) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.java.markdown:132]: add link to /csharp/server/kb/javascript-engine#predefined-javascript-functions

File: Documentation/4.1/Raven.Documentation.Pages/client-api/operations/patching/single-document.java.markdown [hunk 3]
```diff
@@ -123,6 +125,27 @@ An operations interface that exposes the full functionality and allows performin
 
 {PANEL/}
 
+{PANEL: List of Script Methods}
+
+This is a list of a few of the javascript methods that can be used in patch scripts.  
+
```
Comment (garayx) [inline Documentation/4.1/Raven.Documentation.Pages/client-api/operations/patching/single-document.java.markdown:131]: add link to /csharp/server/kb/javascript-engine#predefined-javascript-functions

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 4]
```diff
@@ -128,6 +131,23 @@ An operations interface that exposes the full functionality and allows performin
 
 {PANEL/}
 
+{PANEL: List of Script Methods}
+
+This is a list of a few of the javascript methods that can be used in patch scripts.  
+
```
Comment (garayx) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:137]: add link to /csharp/server/kb/javascript-engine#predefined-javascript-functions

File: Documentation/4.2/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+
+{INFO: PatchOperation}
+
+| Constructor|  | |
+|--------|:-----|-------------| 
+| **id** | `string` | ID of the document to be patched. |
+| **changeVector** | `string` | [Can be null] Change vector of the document to be patched, used to verify that the document was not changed before the patch reached it. |
+| **patch** | `PatchRequest` | Patch request to be performed on the document. |
+| **patchIfMissing** | `PatchRequest` | [Can be null] Patch request to be performed if no document with the given ID was found. Will run only if no `changeVector` was passed. |   
+| **skipPatchIfChangeVectorMismatch** | `bool` | If false and `changeVector` has value, and document with that ID and change vector was not found, will throw exception. |
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:129]: remove redundant column 

image

File: Documentation/4.2/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+| **skipPatchIfChangeVectorMismatch** | `bool` | If false and `changeVector` has value, and document with that ID and change vector was not found, will throw exception. |   
+
+{INFO/}
+
+{PANEL/}
+
+{PANEL: List of Script Methods}
+
+This is a list of a few of the javascript methods that can be used in patch scripts.  
+
```
Comment (garayx) [inline Documentation/4.2/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:138]: add link to /csharp/server/kb/javascript-engine#predefined-javascript-functions

File: Documentation/5.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+
+{INFO: PatchOperation}
+
+| Constructor|  | |
+|--------|:-----|-------------| 
+| **id** | `string` | ID of the document to be patched. |
+| **changeVector** | `string` | [Can be null] Change vector of the document to be patched, used to verify that the document was not changed before the patch reached it. |
+| **patch** | `PatchRequest` | Patch request to be performed on the document. |
+| **patchIfMissing** | `PatchRequest` | [Can be null] Patch request to be performed if no document with the given ID was found. Will run only if no `changeVector` was passed. |   
+| **skipPatchIfChangeVectorMismatch** | `bool` | If false and `changeVector` has value, and document with that ID and change vector was not found, will throw exception. |
```
Comment (garayx) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:129]: remove empty column

File: Documentation/5.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+| **skipPatchIfChangeVectorMismatch** | `bool` | If false and `changeVector` has value, and document with that ID and change vector was not found, will throw exception. |   
+
+{INFO/}
+
+{PANEL/}
+
+{PANEL: List of Script Methods}
+
+This is a list of a few of the javascript methods that can be used in patch scripts.  
+
```
Comment (garayx) [inline Documentation/5.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:138]: add link to /csharp/server/kb/javascript-engine#predefined-javascript-functions

File: Documentation/4.2/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 5 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:129]: Fixed

File: Documentation/5.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 7 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:129]: Fixed

File: Documentation/4.1/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 1 above)
Comment (morhilai) [reply Documentation/4.1/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:136]: Done

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.java.markdown (hunk 2 above)
Comment (morhilai) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.java.markdown:132]: Done

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 4 above)
Comment (morhilai) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:137]: Done

File: Documentation/4.1/Raven.Documentation.Pages/client-api/operations/patching/single-document.java.markdown (hunk 3 above)
Comment (morhilai) [reply Documentation/4.1/Raven.Documentation.Pages/client-api/operations/patching/single-document.java.markdown:131]: Done

File: Documentation/4.2/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 6 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:138]: Done

File: Documentation/5.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 8 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown:138]: Done

**Pull request merged**